web/src/main/java/com/bobocode/bring/web/servlet/DispatcherServlet.java

```diff
@@ -1,44 +1,203 @@
+                        prepareArgs(req, resp, requestPath, method, args);
+                        return method.invoke(bringServlet, args);
+                    }
+                }
+            }
+        }
+        return String.format("This application has no explicit mapping for '%s'", requestPath);
+    }
 
-        resp.addHeader("Trace-Id", id.get());
+    private boolean checkIfPathVariableAnnotationIsPresent(Parameter[] parameters) {
+        for (Parameter parameter : parameters) {
+            if (parameter.isAnnotationPresent(PathVariable.class)) {
+                return true;
+            }
+        }
+        return false;
+    }
 
-        var writer = resp.getWriter();
-        writer.println(mapAsString);
-        writer.flush();
+    @SneakyThrows
+    private void prepareArgs(HttpServletRequest req, HttpServletResponse resp,
+                             String requestPath, Method method, Object[] args) {
+        Parameter[] parameters = method.getParameters();
+        for (int i = 0; i < parameters.length; i++) {
+            if (parameters[i].isAnnotationPresent(PathVariable.class)) {
+                int index = requestPath.lastIndexOf("/");
+                String pathVariable = requestPath.substring(index + 1);
+                Class<?> type = parameters[i].getType();
+                args[i] = parseToParameterType(pathVariable, type);
+            } else if (parameters[i].isAnnotationPresent(RequestParam.class)) {
+                List<String> parameterNames = ReflectionUtils.getParameterNames(method);
+                String parameterName = parameterNames.get(i);
+                Class<?> type = parameters[i].getType();
+                String parameterValue = req.getParameter(parameterName);
+                if (parameterValue != null) {
+                    args[i] = parseToParameterType(parameterValue, type);
+                } else {
+                    throw new MissingServletRequestParameterException(
+                            String.format("Required request parameter '%s' "
+                                            + "for method parameter type '%s' is not present",
+                                    parameterName, type.getSimpleName()));
+                }
+            } else if (parameters[i].isAnnotationPresent(RequestBody.class)) {
+                Class<?> type = parameters[i].getType();
+                if (type.equals(String.class)) {
+                    args[i] = req.getReader().lines().collect(Collectors.joining());
+                } else {
+                    args[i] = objectMapper.readValue(req.getReader(), type);
+                }
+            } else if (parameters[i].getType().equals(HttpServletRequest.class)) {
+                args[i] = req;
+            } else if (parameters[i].getType().equals(HttpServletResponse.class)) {
+                args[i] = resp;
+            }
+        }
+    }
+
+    private Object parseToParameterType(String pathVariable, Class<?> type) {
```

Comment:
It will be good to move to Utils class and we could cover this part in isolation with test if needed.